GLMakie/src/screen.jl

```diff
@@ -387,34 +391,14 @@ function Screen(;
     return screen
 end
 
-function global_gl_screen(resolution::Tuple, visibility::Bool, tries = 1)
-    # ugly but easy way to find out if we create new screen.
-    # could just be returned by global_gl_screen, but dont want to change the API
-    isold = isassigned(GLOBAL_GL_SCREEN) && isopen(GLOBAL_GL_SCREEN[])
-    screen = global_gl_screen()
-    empty!(screen)
-    GLFW.set_visibility!(to_native(screen), visibility)
-    resize!(screen, resolution...)
-    new_size = windowsize(to_native(screen))
-    # I'm not 100% sure, if there are platforms where I'm never
-    # able to resize the screen (opengl might just allow that).
-    # so, we guard against that with just trying another resize one time!
-    if (new_size != resolution) && tries == 1
-        # resize failed. This may happen when screen was previously
-        # enlarged to fill screen. WE NEED TO DESTROY!! (I think)
-        destroy!(screen)
-        # try again
-        return global_gl_screen(resolution, visibility, 2)
-    end
-    # show loading image on fresh screen
-    isold || display_loading_image(screen)
-    screen
-end
```

Comment:
Is there anything important here that should be added to `Screen()`?

Reply:
The if block in this function caused GLMakie tests to fail when you have a space-filling window manager (e.g. `i3` or `bspwm`) installed, cf. https://github.com/JuliaPlots/Makie.jl/issues/1443

Will this be avoided with your rework?

Reply:
Hm, I remember coming up with this horrible code when trying to fix some horrible windows / linux bugs ^^ 
I'd love to get rid of this, but we should make sure it works on all platforms